hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerRpcConnection.java

```diff
@@ -348,8 +348,14 @@ public void saslReadAndProcess(ByteBuff saslToken) throws IOException,
       byte[] replyToken;
       try {
         if (saslServer == null) {
-          saslServer =
+          try {
+            saslServer =
               new HBaseSaslRpcServer(provider, rpcServer.saslProps, rpcServer.secretManager);
+          } catch (Exception e){
```

Comment:
nit: this can be IOException? Exception is also fine though since the purpose is to log any uncaught exceptions.

Reply:
Yeah, the goal here is to catch any uncaught exception that may be thrown by faulty custom providers.